aggregator/src/aggregator.rs

```diff
@@ -2336,33 +2336,18 @@ impl VdafOps {
                     }
 
                     // Write report shares, and ensure this isn't a repeated report aggregation.
-                    // TODO(#225): on repeated aggregation, verify input share matches previously-received input share
                     try_join_all(report_share_data.iter_mut().map(|rsd| {
                         let task = Arc::clone(&task);
-                        let aggregation_job = Arc::clone(&aggregation_job);
 
                         async move {
-                            let put_report_share_fut = tx
-                                .put_scrubbed_report(task.id(), &rsd.report_share)
-                                .or_else(|err| async move {
-                                    match err {
-                                        datastore::Error::MutationTargetAlreadyExists => Ok(()),
-                                        _ => Err(err),
-                                    }
-                                });
-                            // Verify that we haven't seen this report ID and aggregation parameter
-                            // before in another aggregation job.
-                            let report_aggregation_exists_fut = tx
-                                .check_other_report_aggregation_exists::<SEED_SIZE, A>(
-                                    task.id(),
-                                    rsd.report_share.metadata().id(),
-                                    aggregation_job.aggregation_parameter(),
-                                    aggregation_job.id(),
-                                );
-                            let (_, report_aggregation_exists) =
-                                try_join!(put_report_share_fut, report_aggregation_exists_fut)?;
-
-                            if report_aggregation_exists {
+                            let report_already_aggregated =
+                                match tx.put_scrubbed_report(task.id(), &rsd.report_share).await {
+                                    Ok(()) => false,
+                                    Err(datastore::Error::MutationTargetAlreadyExists) => true,
+                                    Err(err) => Err(err)?,
```

Comment:
I think clippy prefers this
```suggestion
                                    Err(err) => return Err(err),
```

Reply:
Clippy is happy enough either way, but I think the explicit `return` makes it clearer no type-coercion magic is happening & (more obviously) that an early return is definitely happening.